src/test/java/com/es/phoneshop/model/cart/DefaultCartServiceTest.java

```diff
@@ -0,0 +1,48 @@
+package com.es.phoneshop.model.cart;
+
+import com.es.phoneshop.model.exceptions.OutOfStockException;
+import com.es.phoneshop.model.product.ArrayListProductDao;
+import com.es.phoneshop.model.product.Product;
+import com.es.phoneshop.model.product.ProductDao;
+import jakarta.servlet.http.HttpServletRequest;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+
+import java.math.BigDecimal;
+import java.util.Currency;
+
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class DefaultCartServiceTest {
```

Comment:
Please add tests to check different scanarios - product added firstly, additional quantity of produc added to cart